rosa-hypershift/rosa-hosted-wrapper.py

```diff
@@ -724,22 +750,58 @@ def _wait_for_workers(kubeconfig, worker_nodes, wait_time, cluster_name):
             time.sleep(15)
             continue
         nodes = nodes_json['items'] if 'items' in nodes_json else []
-        status = []
-        for node in nodes:
-            conditions = node['status']['conditions'] if 'status' in node and 'conditions' in node['status'] else []
-            for condition in conditions:
-                if 'type' in condition and condition['type'] == 'Ready':
-                    status.append(condition['status'])
-        status_list = {i: status.count(i) for i in status}
-        ready_nodes = status_list['True'] if 'True' in status_list else 0
+
+        # First we find nodes which label nodePool match the machinepool name and then we check if type:Ready is on the conditions
+        ready_nodes = sum(
+            len(list(filter(lambda x: x.get('type') == 'Ready' and x.get('status') == "True", node['status']['conditions'])))
+            for node in nodes
+            if node.get('metadata', {}).get('labels', {}).get('hypershift.openshift.io/nodePool')
+            and machinepool_name in node['metadata']['labels']['hypershift.openshift.io/nodePool']
+        ) if nodes else 0
+
         if ready_nodes == worker_nodes:
-            logging.info("Found %d Ready nodes on cluster %s. Expected: %d. Stopping wait." % (ready_nodes, cluster_name, worker_nodes))
-            return ready_nodes
+            logging.info("Found %d/%d ready nodes on machinepool %s for cluster %s. Stopping wait." % (ready_nodes, worker_nodes, machinepool_name, cluster_name))
+            result.append(ready_nodes)
+            result.append(str(datetime.datetime.utcnow().timestamp() - starting_time))
+            return result
         else:
-            logging.info("Found %d Ready nodes on cluster %s. Expected: %d. Waiting 15 seconds for next check..." % (ready_nodes, cluster_name, worker_nodes))
+            logging.info("Found %d/%d ready nodes on machinepool %s for cluster %s. Waiting 15 seconds for next check..." % (ready_nodes, worker_nodes, machinepool_name, cluster_name))
             time.sleep(15)
-    logging.error("Waiting time expired. After %d minutes there are %d ready nodes (Expected: %d) on cluster %s" % (wait_time, ready_nodes, worker_nodes, cluster_name))
-    return ready_nodes
+    logging.error("Waiting time expired. After %d minutes there are %d/%d ready nodes on %s machinepool for cluster %s" % (wait_time, ready_nodes, worker_nodes, machinepool_name, cluster_name))
+    result.append(ready_nodes)
+    result.append("")
+    return result
+
+
+def _add_machinepools(rosa_cmnd, kubeconfig, metadata, machinepool_name, machinepool_flavour, machinepool_labels, machinepool_taints, machinepool_replicas):
+    logging.info('Creating %d machinepools %s-ID on %s, one per AWS Zone' % (len(metadata['zones']), machinepool_name, metadata['cluster_name']))
```

Comment:
Just curious, what is the reason behind one AZ at a time ? why not`--multi-availability-zone`.

Reply:
That parameter is deprecated, but still there on the help

Reply:
ahh, so is `multi-az` going to be default like in `rosa create cluster`  or we should always do zone by zone ?

Reply:
so multi-az is going to be default on cluster, but not on machinepools, each machinepool you add is based on a single AZ, and the default one, workers, will be created on each subnet detected on the installation command.

that is what I understood from @dry923 when we talked about it

Reply:
gotcha, otherwise LGTM

Reply:
always zone by zone